.github/workflows/main.yaml

```diff
@@ -16,30 +16,27 @@ jobs:
         use-verbose-mode: 'yes'
         file-path: './README.md'
     
-  testing:
+  formatting:
     runs-on: ubuntu-22.04
-    name: Run test suites
+    name: Check formatting
     steps:
     - uses: actions/checkout@v3.1.0
     - name: Install nix
       uses: cachix/install-nix-action@v18
       with:
         extra_nix_config: |
           access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
-    - name: Processes backend test
-      run: nix develop .#lightweight --command cabal test smtlib-backends-process
-    - name: Z3 backend test
-      run: nix develop .#lightweight --command cabal test smtlib-backends-z3
+    - run: nix flake check
 
-  formatting:
+  testing:
     runs-on: ubuntu-22.04
-    name: Check formatting
+    name: Run test suites
     steps:
     - uses: actions/checkout@v3.1.0
     - name: Install nix
       uses: cachix/install-nix-action@v18
       with:
         extra_nix_config: |
           access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
-    - run: |
-        nix shell nixpkgs#ormolu --command ormolu --mode check $(find . -name '*.hs')
+    - name: Test backends
+      run: nix develop .#lightweight --command cabal test all
```

Comment:
Wouldn't it be better if this was also run as a flake check, e.g. `nix flake check .#cabal-tests` ?

Reply:
I see. There's probably a way to go around this using a temporary directory during the build but it might be overkill.